network-config-analyzer/NetworkConfig.py

```diff
@@ -358,9 +382,13 @@ def _allowed_xgress_conns(self, from_peer, to_peer, is_ingress):
                                      all_allowed_conns=allowed_conns | allowed_non_captured_conns)
 
         allowed_non_captured_conns = ConnectionSet()
-        if not policy_captured:
-            if self.type in [NetworkConfig.ConfigType.K8s, NetworkConfig.ConfigType.Unknown]:
-                allowed_non_captured_conns = ConnectionSet(True)  # default Allow-all ingress in k8s or in case of no policy
+        if not policy_captured or (self.type == NetworkConfig.ConfigType.Ingress and not is_ingress):
+            if self.type in [NetworkConfig.ConfigType.K8s, NetworkConfig.ConfigType.Ingress,
+                             NetworkConfig.ConfigType.Unknown]:
+                # default Allow-all (not denied by ingress) in k8s or in case of no policy
+                all_conns = ConnectionSet()
+                all_conns.add_all_connections_of_protocol('TCP')
+                allowed_non_captured_conns = all_conns - ingress_denied_conns
             else:
                 if self.profiles:
                     allowed_non_captured_conns = self._get_profile_conns(from_peer, to_peer, is_ingress).allowed_conns
```

Comment:
Right before the return statement, add the following (so it applies to both k8s and calico policies allowed connections) :

```
        if consider_ingress_resource_policies:
            allowed_conns -= ingress_denied_conns
            allowed_non_captured_conns -= ingress_denied_conns
```

Reply:
Do we need to also update `denied_conns`?

Reply:
It's enough that ingress impacts allowed_conns. We don't want to mix the denied_conns of the network policy by ingress.